tests/integration/pilot/istioctl_test.go

```diff
@@ -172,3 +176,77 @@ func getPodID(i echo.Instance) (string, error) {
 
 	return "", fmt.Errorf("no workloads")
 }
+
+func TestProxyConfig(t *testing.T) {
+	framework.NewTest(t).
+		RequiresEnvironment(environment.Kube).
+		RunParallel(func(ctx framework.TestContext) {
+			ns := namespace.NewOrFail(ctx, ctx, namespace.Config{
+				Prefix: "istioctl-pc",
+				Inject: true,
+			})
+
+			var a echo.Instance
+			echoboot.NewBuilderOrFail(ctx, ctx).
+				With(&a, echoConfig(ns, "a")).
+				BuildOrFail(ctx)
+
+			istioCtl := istioctl.NewOrFail(t, ctx, istioctl.Config{})
+
+			podID, err := getPodID(a)
+			if err != nil {
+				ctx.Fatalf("Could not get Pod ID: %v", err)
+			}
+
+			var output string
+			var args []string
+			g := gomega.NewGomegaWithT(t)
+
+			args = []string{fmt.Sprintf("--namespace=%s", ns.Name()),
+				"pc", "bootstrap", podID}
+			output = istioCtl.InvokeOrFail(t, args)
+			jsonOutput := jsonUnmarshallOrFail(t, strings.Join(args, " "), output)
+			g.Expect(jsonOutput).To(gomega.HaveKey("bootstrap"))
+
+			args = []string{fmt.Sprintf("--namespace=%s", ns.Name()),
+				"pc", "cluster", podID, "-o", "json"}
+			output = istioCtl.InvokeOrFail(t, args)
+			jsonOutput = jsonUnmarshallOrFail(t, strings.Join(args, " "), output)
+			g.Expect(jsonOutput).To(gomega.Not(gomega.BeEmpty()))
+
+			args = []string{fmt.Sprintf("--namespace=%s", ns.Name()),
+				"pc", "endpoint", podID, "-o", "json"}
+			output = istioCtl.InvokeOrFail(t, args)
+			jsonOutput = jsonUnmarshallOrFail(t, strings.Join(args, " "), output)
+			g.Expect(jsonOutput).To(gomega.Not(gomega.BeEmpty()))
```

Comment:
Is the aim here just to test that we dont get version mismatches? I _think_ this is all this will catch.

Reply:
All it catches is that the tool can find the proper config section even if the version/types change.

With the upgrade to Envoy v3 the `istioctl pc` on Master no longer works with Istio 1.4.

This test won't catch that kind of drift, but it will check that we can at least parse.  All of the `pc` subcommands do a lot of work extracting and unmarshalling that is version specific.  This test will make sure that we get some JSON output, which means that the command finds the correct section when the version changes.

Once this test is in there will be a follow-up PR (I am still working on it) to get the tool to handle both v3 and v2alpha Envoy JSON.